### ISSUE TITLE
New check: CommentAttachedCheck flag blank line between a comment/Javadoc and the declaration it precedes

### ISSUE THREAD

**Issue by ZaheerAhmadDev** [checkstyle/checkstyle#21682]
I have read check documentation:https://checkstyle.sourceforge.io/checks.html
I have downloaded the latest cli from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

**How the new check implimentation should works:**

**javac:**

```java
zaheerahmad@fedora:~/checks$  cat src/Test.java
class Test {

    /**
     * some comment
     */

     public void javadoc() {}


    // some comment explaining check

     public void singlelinecomment() {}

    /* This is a multi-line comment.
       It spans across multiple lines
        without breaking the code. */

     public void multilinecomment() {}

}
zaheerahmad@fedora:~/checks$ javac src/Test.java
zaheerahmad@fedora:~/checks$ 
```
**Example with violation Expected:**

```xml
zaheerahmad@fedora:~/checks$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
  "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
  "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="CommentAttachedCheck"/>
  </module>
</module>
```

```java
zaheerahmad@fedora:~/checks$  cat src/Test.java
class Test {

    /**
     * some comment
     */

     public void javadoc() {} // should show of blank line between method and doc


    // some comment explaining check

     public void singlelinecomment() {} // should show of blank line between method and doc

    /* This is a multi-line comment.
       It spans across multiple lines
        without breaking the code. */

     public void multilinecomment() {} // should show of blank line between method and doc

}
```

```
zaheerahmad@fedora:~/checks$ java -jar checkstyle-14.1.0-all.jar -c config.xml src/Test.java
Starting audit...
[ERROR] /home/zaheerahmad/checks/src/Test.java:7:6:'METHOD_DEF' should not be separated from its comment by an empty line.. [CommentAttachedCheck]
[ERROR] /home/zaheerahmad/checks/src/Test.java:12:6:'METHOD_DEF' should not be separated from its comment by an empty line. [CommentAttachedCheck]
[ERROR] /home/zaheerahmad/checks/src/Test.java:18:6:'METHOD_DEF' should not be separated from its comment by an empty line. [CommentAttachedCheck]
Audit done.
Checkstyle ends with 3 errors.
```

**Is your feature request related to a problem? Please describe.**
Checkstyle currently has no way to flag a comment  `Javadoc, block comment,
or line comment` that is visually detached from the declaration it appears to document. i think we should avoid blank line between comment and it declaration.

**Describe the solution you'd like**
we can introduce new check  somthing like`CommentAttachedCheck` that check for blank between `comment and javadoc` and it declaration.

**Example with no violation is expected**:
```java
zaheerahmad@fedora:~/checks$  cat src/Test.java
class Test {

    /**
     * some comment
     */
     public void javadoc() {} //ok as there is blankline

    // some comment explaining check
     public void singlelinecomment() {} //ok as there is blankline

    /* This is a multi-line comment.
       It spans across multiple lines
        without breaking the code. */
     public void multilinecomment() {} //ok as there is blankline

}
```
```zaheerahmad@fedora:~/checks$ java -jar checkstyle-14.1.0-all.jar -c config.xml src/Test.java
Starting audit...
Audit done.
```

**Comment by ZaheerAhmadDev** [comment]
@romani Please share your opinion. Is it okay to allow a blank line between a method and its comment or Javadoc.
